hostsfile/src/lib.rs

```diff
@@ -206,9 +210,39 @@ impl HostsBuilder {
         let begin = lines.iter().position(|line| line.trim() == begin_marker);
         let end = lines.iter().position(|line| line.trim() == end_marker);
 
+        let mut lines_to_insert = vec![];
+        if !self.hostname_map.is_empty() {
+            writeln!(lines_to_insert, "{begin_marker}")?;
```

Comment:
Currently `lines_to_insert` is `Vec<u8>`, but I think it would simplify code below if it were a `Vec<String>`. I.e. we would be pushing lines as Strings to the Vec as opposed to writing it to a continuous buffer.

It should make the comparison with `lines` below more straightforward.

Reply:
This definitely seems like the best suggestion for simplifying the code here.